### DISCUSSION TITLE
Introducing sampling method for Kepler and finding reasonable sampling rate

### DISCUSSION THREAD

**Discussion by eklee15** [sustainable-computing-io/kepler#836] [Ideas]
Due to the high overhead of Kepler/eBPF, we are investigating the sampling method for Kepler. 
https://github.com/sustainable-computing-io/kepler/pull/823#issuecomment-1653498569

However, because sampling would potentially reduce the accuracy of the model, we would want to find a good sampling rate for Kepler. Of course, this would depend on workload, application, and system. This tradeoff study would be very helpful for Kepler in the longer term.

**Comment by rootfs** [comment]
https://github.com/sustainable-computing-io/kepler/pull/1685
